### ISSUE TITLE
Bug: unable to see the components in the components tab

### ISSUE THREAD

**Issue by plegault3397** [facebook/react#21175]
Installed the react development tools to chrome Version 89.0.4389.114 (Official Build) (64-bit)
Open my development version of my react app localhost:3000 with several components and I see the react icon
![image](https://user-images.githubusercontent.com/7073595/113520660-a1463a80-9562-11eb-9b6d-82919344c146.png)
Go to the inspect and component to and nothing is showing
![image](https://user-images.githubusercontent.com/7073595/113520674-ccc92500-9562-11eb-87d0-5068abc150e1.png)

I'm using this chode https://codesandbox.io/s/react-devtools-dk9gj?fontsize=14
and when you open this in chrome it works.
I open it from localhost and it does not work.
![image](https://user-images.githubusercontent.com/7073595/113520835-d2733a80-9563-11eb-88cb-7bb6d6b30ade.png)

    "react": "16.8.6",
    "react-dom": "16.8.6",
    "react-scripts": "^4.0.3"
npm --version 7.6.3
nodejs v14.16.0

## Steps To Reproduce

1.
2.

Link to code example:

**Comment by bvaughn** [comment]
The Code Sandbox code you linked to works, so what are the steps to reproduce the "does not work" part?

**Comment by plegault3397** [comment]
fork it locally and download it to your pc run the npm i. then try with localhost:3000

**Comment by bvaughn** [comment]
> fork it locally and download it to your pc run the npm i. then try with localhost:3000

Works fine for me.

Fresh install of the React extension. All default options.

**Comment by plegault3397** [comment]
Still doesn't work, I've uninstalled re-installed rebooted several times.
works in code sandbox fine, just not on my local sites.
Maybe my installation of react, the icon looks the same in both my local web and code sandbox.
I don't get it!

**Comment by deflock** [comment]
I can confirm this. I've just recreated chrome profile from scratch and installed only React Devtools extension.
And Components tab is empty.
I suspect that the reason is the new major version 4.x of `react-scripts`.

**Comment by bvaughn** [comment]
@deflock Can you share a repo that I can check out and run and reproduce the problem?

Sounds like maybe Fast Refresh integration is breaking stuff somehow but I don't know how to repro.

**Comment by deflock** [comment]
@bvaughn I can't share my current affected project, but I'll try to reproduce this from scratch if I'll have enough time for this

**Comment by bvaughn** [comment]
If you can share something, that will be useful. Without a repro, this issue will eventually be closed.

**Comment by deflock** [comment]
@bvaughn Finally had some time for this.

https://github.com/deflock/devtools-empty-components
Reproduced with this. Really simple. Maybe it's Windows-related.

**Comment by bvaughn** [comment]
Thanks Alex! I'll try to take a look Monday

**Comment by bvaughn** [comment]
Looking at this app, DevTools isn't getting any messages from React that an app has mounted. This makes me wonder if something is sitting between React and the `__REACT_DEVTOOLS_GLOBAL_HOOK__` (the thing DevTools uses to subscribe to changes).

**Comment by bvaughn** [comment]
I think this is a duplicate of #20100.

Fast Refresh is inserting itself between React and React DevTools in DEV mode and bailing out:
```js
hook.onCommitFiberRoot = function (id, root, maybePriorityLevel, didError) {
  var helpers = helpersByRendererID.get(id);

  if (helpers === undefined) {
    return; // This is happening
  }

  // ...
```

This is happening because the thing being _injected_ doesn't define `scheduleRefresh` or `scheduleRefresh` functions, so Fast Refresh doesn't map it:
https://github.com/facebook/react/blob/9d48779b362ef773da50fef95af6998b560c75c0/packages/react-refresh/src/ReactFreshRuntime.js#L494-L504

Then the commit hook later bails out because there's no helper:
https://github.com/facebook/react/blob/dfb6a403357d5ac2c0ea87aa6e5a4f947de342f8/packages/react-refresh/src/ReactFreshRuntime.js#L513-L522

That being said, this has already been fixed (#20129) and Fast Refresh no longer bails out in this case:
https://github.com/facebook/react/blob/9d48779b362ef773da50fef95af6998b560c75c0/packages/react-refresh/src/ReactFreshRuntime.js#L494-L504

I think this means that the version of Fast Refresh that's running just doesn't include my fix from #20129. Looking at the `package-log.json` file, I see `react-refresh` version 0.8.3 is being used. Looks like the latest version of 0.10.0. Installing that version to your project fixes the problem:
```sh
npm i react-refresh@^0.10.0 
```

Going to close this since the bug here was already fixed!

**Comment by plegault3397** [comment]
Thank you
